flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/PhysicalTableSourceScan.scala

```diff
@@ -70,4 +73,18 @@ abstract class PhysicalTableSourceScan(
 
   def copy(traitSet: RelTraitSet, tableSource: TableSource[_]): PhysicalTableSourceScan
 
+  override def estimateRowCount(mq: RelMetadataQuery): Double = {
+    val tableSourceTable = getTable.unwrap(classOf[TableSourceTable[_]])
+
+    if (tableSourceTable.getStatistic != FlinkStatistic.UNKNOWN) {
```

Comment:
The basic idea of current accessing order is let each ``TableSource`` decide its statistic (Only it knows how to get correct statistic. And  in future there are different kinds of ``XXXTableSource`` ).

we can change the result value of ``fromExternalCatalogTable`` method in ``TableSourceConverter``:

``
def fromExternalCatalogTable(externalCatalogTable: ExternalCatalogTable): (T, TableStats)
``

Currently, if a table source is ``FilterableTableSource``, the returned ``TableStats`` is null.